libevmasm/ConstantOptimiser.cpp

```diff
@@ -263,6 +266,24 @@ bool ComputeMethod::checkRepresentation(u256 const& _value, AssemblyItems const&
 			case Instruction::NOT:
 				sp[0] = ~sp[0];
 				break;
+			case Instruction::SHL:
+				assertThrow(
+					m_params.evmVersion.hasBitwiseShifting(),
+					OptimizerException,
+					"Shift generated for invalid EVM version."
+				);
+				assertThrow(sp[0] <= u256(255), OptimizerException, "Invalid shift generated.");
+				sp[-1] = u256(bigint(sp[-1]) << unsigned(sp[0]));
+				break;
+			case Instruction::SHR:
+				assertThrow(
+					m_params.evmVersion.hasBitwiseShifting(),
+					OptimizerException,
+					"Shift generated for invalid EVM version."
+				);
+				assertThrow(sp[0] <= u256(255), OptimizerException, "Invalid shift generated.");
```

Comment:
Should all these assertions just return false (as `EXP` is doing)?

Reply:
Hm, returning false will lead to an OptimizerException. Not sure what is better.